search/src/test/scala/weco/api/search/images/ImagesSimilarityTest.scala

```diff
@@ -1,85 +1,126 @@
 package weco.api.search.images
 
 class ImagesSimilarityTest extends ApiImagesTestBase {
-  it("includes visually similar images with ?include=visuallySimilar") {
-    withImagesApi {
-      case (imagesIndex, routes) =>
-        indexTestDocuments(
-          imagesIndex,
-          (0 to 5).map(i => s"images.similar-features-and-palettes.$i"): _*
-        )
 
-        assertJsonResponse(
-          routes,
-          path =
-            s"$rootPath/images/${getTestImageId("images.similar-features-and-palettes.0")}?include=visuallySimilar"
-        ) {
-          Status.OK -> readResource(
-            "expected_responses/visually-similar-features-and-palettes.json"
+  /**
+    * These tests treat the simple presence of the requested property as a proxy
+    * for populating the similar image lists.  Essentially, this is showing the
+    * connection between the `include` querystring parameter and the resulting JSON.
+    *
+    * The thresholds are set too high to be able to actually return similar images in
+    * these dummy indices, so the content of these image lists is irrelevant to the tests.
+    *
+    * Tests that show that the similar images functionality behaves as expected
+    * can be found in ../services/ImagesServiceTest.scala
+    * and there should also be rank tests to cover whether the images returned are appropriately similar
+    */
```

Comment:
This is a good comment; I wonder if it would be leaning on these tests to check some of the new kNN behaviour is working correctly? i.e. that images are never similar to themselves

While I was testing locally, I commented out the code in ImagesService that filters out an image from the kNN response, but this test still passes.

So the assertions become something like "the JSON contains `visuallySimilar` _and_ it's an empty list".

Reply:
I think that's a good plan.  These tests originally loaded half a dozen similar records, then checked whether the list for one contained all the others.  When the thresholds changed some months ago, this was still loading them all but now checking for an empty list.

I thought that would be fragile, as the random fuzz in two similar records could possibly result in documents similar enough to get into the list.

Then I removed all the redundant source records, but didn't make the leap that this meant the test could now rely on the lists to be empty.